dds/idl/marshal_generator.cpp

```diff
@@ -389,7 +389,40 @@ namespace {
     return arg.str();
   }
 
-  void gen_sequence(UTL_ScopedName* tdname, AST_Sequence* seq)
+  // Helper function to conditionally generate dheader-related code which is passed as the
+  // "code" argument.
+  // may_be_delimited determines if the dheader is ever needed for this type.
+  // not_only_delimited determines if the dheader code is conditional based on the encoding.
+  // code contains the lines of code to generate
+  // is_ser_func specifies whether this is a serialization operation
+  void generate_dheader_code(bool may_be_delimited, bool not_only_delimited,
+                             const std::vector<std::string>& code, bool is_ser_func = true)
+  {
+    if (may_be_delimited) {
```

Comment:
Would it be more readable for the calling functions to have this conditional check and not call this function if it's not needed?

Reply:
That would lead to more repeated code.  I think it is fine the way it is as the dheader would only apply if it could be delimited.